effects/sleep.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import time
+
+import click
+
+from rpi_ws281x import PixelStrip, Color
+
+# LED strip default configuration:
+#LED_COUNT = 50       # Number of LED pixels.
+LED_PIN = 18          # GPIO pin connected to the pixels (18 uses PWM!).
+LED_FREQ_HZ = 800000  # LED signal frequency in hertz (usually 800khz)
+LED_DMA = 10          # DMA channel to use for generating signal (try 10)
+LED_BRIGHTNESS = 255  # Set to 0 for darkest and 255 for brightest
+LED_INVERT = False    # True to invert the signal (when using NPN transistor level shift)
+LED_CHANNEL = 0       # set to '1' for GPIOs 13, 19, 41, 45 or 53
+
+@click.command()
+@click.option('-l', '--leds', type=int, default=50, show_default=True, help='Amount of LEDs in the string')
+@click.option('-s', '--speed', type=float, default=1.0, show_default=True, help='Speed of the effect')
+@click.option('-c', '--color', type=str, default='FF0000', show_default=True, help='RGB Value for the effect. You need to provide this value as a hexadecimal.')
+def main(leds: int, speed: float, color: str):
+    '''
+    This is a sleep effect which provide a light for the night
+    '''
+    if not color.startswith('0x'):
+        color = '0x' + color
+        
+    print(f'Sleep effect started with following parameter: leds={leds}, speed={speed}, color={color}')
+    color_dict = parseColorString(color)
+    
+    strip = PixelStrip(leds, LED_PIN, LED_FREQ_HZ, LED_DMA, LED_INVERT, LED_BRIGHTNESS, LED_CHANNEL)
+    strip.begin() # Intialize the library (must be called once before other functions).
+    
+    runEffect(strip, speed, color_dict['red'], color_dict['green'], color_dict['blue']);
+    pass
+
+def parseColorString(color: str):
+    '''
+    Parse the color given as string to single RGB colors as int values and return the dictionary with
+    parsed values
+    '''
+    color_int_val = int(color, base=16)
+    
+    return {
+        'red': (color_int_val >> 16) & 0xFF,
+        'green': (color_int_val >> 8) & 0xFF,
+        'blue': color_int_val & 0xFF
+    }
+
+def runEffect(strip, speed, red_max_val, green_max_val, blue_max_val):
+    SLEEP_TIME = 0.005 / speed
+
+    for pixel_index in range(strip.numPixels()+1):
+        prev_pixel_index = pixel_index - 1 # Index of the pixel we want to fade out
+
+        for brightnes in range(0, 256):
+            brightnes_factor = brightnes / 255.0
+            red_val = int(red_max_val * brightnes_factor)
+            green_val = int(green_max_val * brightnes_factor)
+            blue_val = int(blue_max_val * brightnes_factor)
+
+            if pixel_index < strip.numPixels():
+                strip.setPixelColor(pixel_index, Color(green_val, red_val, blue_val)) # Fade in pixel
```

Comment:
Is the order of colors correct? I.e. green before red instead of the other way round?

Reply:
Yes,it is correct xD ... At least my LED strip shows this correct ... i think if you have another LED strip which shows the effects incorrect, we need to revisit this code and how we can define the color pattern

Reply:
I don't think that's correct. Please have a look at the library:
https://github.com/rpi-ws281x/rpi-ws281x-python/blob/master/library/rpi_ws281x/rpi_ws281x.py#L13
This states that the order of colors is red, green, blue. I think you have some setup where you swapped green and red and it just works a a coincidence.